pilot/cmd/pilot-agent/status/server.go

```diff
@@ -718,10 +718,16 @@ func (s *Server) handleAppProbeHTTPGet(w http.ResponseWriter, req *http.Request,
 		return
 	}
 
-	// Forward incoming headers to the application.
-	if prober.HTTPGet.Host != "" {
-		appReq.Host = prober.HTTPGet.Host
+	appReq.Host = req.Host
+	if host, port, err := net.SplitHostPort(req.Host); err == nil {
+		port, _ := strconv.Atoi(port)
+		// the port is same as the status port, then we need to replace the port in the host with the real one
```

Comment:
I thought about this but was worried about crazy edge case where they actually set the host explicitly to `foobar:15021` and do not want it overwritten...

but that probably doesn't really matter

Reply:
Good catch, have no way to prevent this case. In real, it is almost impossible to have an application set conflict port with status port.